src/models/guide.cpp

```diff
@@ -19,8 +20,121 @@
 namespace {
 Logger logger(LOG_MAIN, "Guide");
 
+bool evaluateConditionsEnabledFeatures(const QJsonArray& enabledFeatures) {
+  for (QJsonValue enabledFeature : enabledFeatures) {
+    // If the feature doesn't exist, we crash.
+    const Feature* feature = Feature::get(enabledFeature.toString());
+    Q_ASSERT(feature);
+
+    if (!feature->isSupported()) {
+      logger.info() << "Feature not supported";
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool evaluateConditionsPlatforms(const QJsonArray& platformArray) {
+  QStringList platforms;
+  for (QJsonValue platform : platformArray) {
+    platforms.append(platform.toString());
+  }
+
+  if (!platforms.isEmpty() &&
+      !platforms.contains(MozillaVPN::instance()->platform())) {
+    logger.info() << "Not supported platform";
+    return false;
+  }
+
+  return true;
+}
+
+bool evaluateConditionsSettingsOp(const QString& op, bool result) {
+  if (op == "eq") return result;
+
+  if (op == "neq") return !result;
+
+  logger.error() << "Invalid settings operator" << op;
```

Comment:
In the taxonomy of logging levels, this should be a warning(): something has gone wrong in an unexpected way but the application can continue to function.